spec/mem/tree_borrows/state_machine.md

```diff
@@ -65,49 +65,43 @@ impl Permission {
         )
     }
 
-    fn child_write(self, protected: Protected) -> Result<Permission> {
+    fn child_write(self, protected: bool) -> Result<Permission> {
         match self {
-            Permission::Reserved { conflicted: true } if protected.yes() =>
+            Permission::Reserved { conflicted: true } if protected =>
                 throw_ub!("Tree Borrows: writing to the child of a protected pointer with Conflicted Reserved permission"),
             Permission::Frozen => throw_ub!("Tree Borrows: writing to the child of a pointer with Frozen permission"),
             Permission::Disabled => throw_ub!("Tree Borrows: writing to the child of a pointer with Disabled permission"),
             _ => ret(Permission::Active),
         }
     }
 
-    fn foreign_read(self, protected: Protected, accessed: Accessed) -> Result<Permission> {
+    fn foreign_read(self, protected: bool) -> Result<Permission> {
         match self {
-            Permission::Active if protected.active(accessed) =>
-                throw_ub!("Tree Borrows: foreign read of an actively protected pointer with Active permission"),
-            Permission::Reserved { .. } if protected.yes() => ret(Permission::Reserved { conflicted: true }),
+            Permission::Active if protected  => ret(Permission::Disabled),
+            Permission::Reserved { .. } if protected => ret(Permission::Reserved { conflicted: true }),
             Permission::Active => ret(Permission::Frozen),
             // All other states are kept unchanged.
             perm => ret(perm),
         }
     }
 
-    fn foreign_write(self, protected: Protected, accessed: Accessed) -> Result<Permission> {
+    fn foreign_write(self, protected: bool) -> Result<Permission> {
         match self {
-            Permission::ReservedIM if !protected.active(accessed) => ret(Permission::ReservedIM),
-            _ if !protected.active(accessed) => ret(Permission::Disabled),
-            Permission::Reserved { .. } => throw_ub!("Tree Borrows: writing to the foreign of an actively protected pointer with Reserved permission"),
-            Permission::Active => throw_ub!("Tree Borrows: writing to the foreign of an actively protected pointer with Active permission"),
-            Permission::Frozen => throw_ub!("Tree Borrows: writing to the foreign of an actively protected pointer with Frozen permission"),
-            Permission::Disabled => panic!("Permission::foreign_write: Impossible Combination (Accessed + Protected + Disabled)"),
-            Permission::ReservedIM => panic!("Permission::foreign_write: Impossible Combination (Accessed + Protected + ReservedIM)"),
+            Permission::ReservedIM if !protected => ret(Permission::ReservedIM),
```

Comment:
The `!protected` is unnecessary. It's still there because it being there makes the theorem that nothing protected survives foreign writes much simpler to prove (it holds by definition)

Reply:
Please add a comment explaining this.